todo_app/data/todoapi.py

```diff
@@ -0,0 +1,160 @@
+from abc import ABC, abstractmethod
+from configparser import ConfigParser
+from .boardelements import *
+import os
+import requests
+
+_TRELLO = 'trello'
+
+def todo_api_factory(api_name):
+    ''' Factory method for creating a concrete ToDoAPI. Currently only accepts 'trello' '''
+    if api_name == _TRELLO:
+        return TrelloAPI()
+    else:
+        #TODO add more APIs in the future
+        return None
+
+
+class ToDoAPI(ABC):
+
+    @abstractmethod
+    def get_list_of_statuses(self):
+        pass
+
+    @abstractmethod
+    def get_list_of_items(self):
+        pass
+
+    @abstractmethod
+    def add_item(self, title):
+        pass
+    
+    @abstractmethod
+    def delete_item(self, id):
+        pass
+
+    @abstractmethod
+    def modify_item(self, id, attributes):
+        pass
+
+    @property
+    def todomapper(self) -> ToDoMapper:
+        return self._todomapper
+
+    @todomapper.setter
+    def todomapper(self, mapper: ToDoMapper):
+        self._todomapper = mapper
+
+class TrelloAPI(ToDoAPI):
+    TRELLO_ROOT_URL = 'https://api.trello.com'
+    CREDS_CONFIG_FILE = 'creds.config'
+   
+    def __init__(self):
+        self.init_todomapper()    
+
+        #keep this call at the top as needed to get authorisation details for all subsequent calls
+        self.auth_query = self.get_trello_creds_section('trello_auth')
+        self.user = self.get_trello_creds_section('trello_user').get('user')
+
+        self.boards = self.get_boards_of_user(self.user) 
+        #Currently defaulting to the first board - in future we can have dropdown
+        self.board_id = self.boards[0].id 
```

Comment:
This might be unexpected (it was for me!). Ideally you could specify the board you want to edit (maybe as a config value)

Reply:
Agree - moved to config file (now renamed to trello.config)